homeassistant/components/broadlink/switch.py

```diff
@@ -54,12 +68,55 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     mac_addr = config.get(CONF_MAC)
     switches = config.get(CONF_SWITCHES)
 
+    if not isinstance(switches, list):
+        switches = [
+            {
+                CONF_NAME: switch[CONF_FRIENDLY_NAME],
+                CONF_COMMAND_OFF: switch[CONF_COMMAND_OFF],
+                CONF_COMMAND_ON: switch[CONF_COMMAND_ON],
```

Comment:
These items are optional in the old schema, so they might be missing.

Reply:
Well remembered. I will have to use the slug as a fallback for the friendly name, something like this:
```python3
if not isinstance(switches, list):
    switches = [
        {
            CONF_NAME: switch.pop(CONF_FRIENDLY_NAME, None) or name,
            **switch,
        }
        for name, switch in switches.items()
    ]
```
We could also use the slug as the main option for the name, so we avoid breaking the entity ids.